packages/graphql/src/utils/unitTokenValue.js

```diff
@@ -0,0 +1,77 @@
+import memoize from 'lodash/memoize'
+import contracts from '../contracts'
+import currencies from './currencies'
+
+/**
+ * Get the ether unit that has a certain number of decimals
+ *
+ * Ref: https://web3js.readthedocs.io/en/v1.2.0/web3-utils.html#id88
+ *
+ * @param {Number} decimals number of decimals
+ *
+ * @returns {String} the ether unit that has `decimals` decimals
+ */
+const getUnitWithDecimals = memoize(decimals => {
+  const web3 = contracts.web3
+
+  const unit = Object.keys(web3.utils.unitMap).find(unit => {
+    const valueInWei = web3.utils.unitMap[unit]
+
+    const zeroCount = valueInWei.length - valueInWei.replace(/0/g, '').length
+
+    return zeroCount === decimals
```

Comment:
AFAICT web3 units for each zero count [don't exist](https://web3js.readthedocs.io/en/v1.2.0/web3-utils.html#id88). Sometimes there are 3 zeros between the 2 closest units. So the above function might miss the correct unit in case its' decimal count is between 2 units. (for example between `szabo` -> 12 zeros and `nano` -> 9 zeros)

Maybe one suggestion as how to amend the above implementation: 
```
let biggestUnit = 'wei'
let c = 0
const unit = Object.keys(web3.utils.unitMap).forEach(unit => {
    const valueInWei = web3.utils.unitMap[unit]

    const zeroCount = valueInWei.length - valueInWei.replace(/0/g, '').length
    if (zeroCount <= decimals && zeroCount > biggestUnitZeros) {
        biggestUnit = unit
        biggestUnitZeros = zeroCount
    }
```

Reply:
Thanks for catching this @sparrowDom, Will fix this